src/GraphQL.Tests/DI/GraphQLBuilderExtensionTests.cs

```diff
@@ -898,6 +898,82 @@ public void ConfigureExecution_Null()
     }
     #endregion
 
+    #region - AddUnhandledExceptionHandler -
+    [Fact]
+    public async Task AddUnhandledExceptionHandler1()
+    {
+        ExecutionOptions? o2 = null;
+        UnhandledExceptionContext? e2 = null;
+        var execute = MockSetupConfigureExecution();
+        _builder.AddUnhandledExceptionHandler((e1, o1) =>
+        {
+            o2 = o1;
+            e2 = e1;
+            return Task.CompletedTask;
+        });
+        var opts = execute();
+        var e = new UnhandledExceptionContext(null, null, new Exception());
+        await opts.UnhandledExceptionDelegate(e);
+        e2.ShouldBe(e);
+        o2.ShouldBe(opts);
+    }
+
+    [Fact]
+    public async Task AddUnhandledExceptionHandler2()
+    {
+        UnhandledExceptionContext? e2 = null;
+        var execute = MockSetupConfigureExecution();
+        _builder.AddUnhandledExceptionHandler(e1 =>
+        {
+            e2 = e1;
+            return Task.CompletedTask;
+        });
+        var opts = execute();
+        var e = new UnhandledExceptionContext(null, null, new Exception());
+        await opts.UnhandledExceptionDelegate(e);
+        e2.ShouldBe(e);
+    }
+
+    [Fact]
+    public async Task AddUnhandledExceptionHandler3()
+    {
+        ExecutionOptions? o2 = null;
+        UnhandledExceptionContext? e2 = null;
+        var execute = MockSetupConfigureExecution();
+        _builder.AddUnhandledExceptionHandler((e1, o1) =>
+        {
+            o2 = o1;
+            e2 = e1;
+        });
+        var opts = execute();
+        var e = new UnhandledExceptionContext(null, null, new Exception());
+        await opts.UnhandledExceptionDelegate(e);
+        e2.ShouldBe(e);
+        o2.ShouldBe(opts);
+    }
+
+    [Fact]
+    public async Task AddUnhandledExceptionHandler4()
+    {
+        UnhandledExceptionContext? e2 = null;
+        var execute = MockSetupConfigureExecution();
+        _builder.AddUnhandledExceptionHandler(e1 => e2 = e1);
+        var opts = execute();
+        var e = new UnhandledExceptionContext(null, null, new Exception());
+        await opts.UnhandledExceptionDelegate(e);
+        e2.ShouldBe(e);
+    }
+
+    [Fact]
+    public void AddUnhandledExceptionHandler_Null()
+    {
+        Should.Throw<ArgumentNullException>(() => _builder.AddUnhandledExceptionHandler((Action<UnhandledExceptionContext>)null!));
+        Should.Throw<ArgumentNullException>(() => _builder.AddUnhandledExceptionHandler((Action<UnhandledExceptionContext, ExecutionOptions>)null!));
+        Should.Throw<ArgumentNullException>(() => _builder.AddUnhandledExceptionHandler((Func<UnhandledExceptionContext, Task>)null!));
+        Should.Throw<ArgumentNullException>(() => _builder.AddUnhandledExceptionHandler((Func<UnhandledExceptionContext, ExecutionOptions, Task>)null!));
```

Comment:
The 4 overloads are shown here.  Basically, synchronous or asynchronous, with or without ExecutionOptions.